services/QuillLMS/engines/evidence/spec/services/evidence/gen_ai/on_topic/checker_spec.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Evidence::GenAI::OnTopic::Checker, type: :service do
+  let(:prompt) { double('prompt') }
+  let(:system_prompt) { 'Mocked system prompt' }
+  let(:entry) { 'Some content' }
+
+  before do
+    allow(Evidence::GenAI::OnTopic::PromptBuilder)
+      .to receive(:run)
+      .with(prompt:)
+      .and_return(system_prompt)
+  end
+
+  subject { described_class.new(prompt:, entry:) }
+
+  describe 'initialization' do
+    it 'sets prompt and entry attributes' do
+      expect(subject.prompt).to eq(prompt)
+      expect(subject.entry).to eq(entry)
+    end
+  end
+
+  describe '#run' do
+    let(:llm_response) { { 'on-topic' => true } }
+
+    before do
+      allow(Evidence::Gemini::Chat)
+        .to receive(:run)
+        .with(
+          system_prompt:,
+          entry:,
+          model: Evidence::Gemini::Chat::SMALL_MODEL
+        )
+        .and_return(llm_response)
+    end
+
+    it { expect(subject.run).to be true }
+
+    context 'negative response' do
+      let(:llm_response) { { 'on-topic' => false } }
+
+      it { expect(subject.run).to be false }
+    end
+  end
+
+  describe 'private methods' do
+    it 'uses the correct chat API model' do
+      expect(subject.send(:model)).to eq(Evidence::Gemini::Chat::SMALL_MODEL)
+    end
+
+    context 'system_prompt' do
+      it 'calls PromptBuilder with the correct prompt' do
+        expect(Evidence::GenAI::OnTopic::PromptBuilder)
+          .to receive(:run)
+          .with(prompt: prompt)
+          .and_return(system_prompt)
+
+        subject.send(:system_prompt)
+      end
+    end
+
+    context 'llm_response' do
+      let(:expected_response) { { 'on-topic' => true } }
+
+      before do
+        allow(Evidence::Gemini::Chat)
+          .to receive(:run)
+          .once
+          .and_return(expected_response)
+      end
+
+      it 'memoizes the LLM response' do
+        2.times { subject.send(:llm_response) }
+      end
```

Comment:
Nice approach here.  I didn't realize that `allow(...).once` would error if you tried to call it more than once.